pkg/controller/migcluster/storageclass.go

```diff
@@ -0,0 +1,74 @@
+package migcluster
+
+import (
+	"strconv"
+
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+	kapi "k8s.io/api/core/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Sets the list of storage classes from the cluster
+func (r *ReconcileMigCluster) setStorageClasses(cluster *migapi.MigCluster) error {
+	var client k8sclient.Client
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+	clusterStorageClasses, err := cluster.GetStorageClasses(client)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+	var storageClasses []migapi.StorageClass
+	for _, clusterStorageClass := range clusterStorageClasses {
+		storageClass := migapi.StorageClass{
+			Name:        clusterStorageClass.Name,
+			Provisioner: clusterStorageClass.Provisioner,
+			AccessModes: r.accessModesForProvisioner(clusterStorageClass.Provisioner),
+		}
+		if clusterStorageClass.Annotations != nil {
+			storageClass.Default, _ = strconv.ParseBool(clusterStorageClass.Annotations["storageclass.kubernetes.io/is-default-class"])
+		}
+		storageClasses = append(storageClasses, storageClass)
+	}
+	cluster.Spec.StorageClasses = storageClasses
+	return nil
+}
+
+var accessModeList = map[string][]kapi.PersistentVolumeAccessMode{
```

Comment:
I think we need to add it because we're only including those provisioners we know about. If a user has other third-party provisioners installed that we don't know about, we assume RWO only, so if they have, for example, NFS provisioners that support RWX, we'll report warnings. Now we're intentionally only warning and not causing an error condition, so they could just document for their users to ignore the warning, but it would be better if we could allow them to modify this with a configmap. If we add that, though, it should be a follow-on PR rather than holding up the base functionality here.

Reply:
Hi @sseago and @dymurray  - Our provisioner name is  csi.trident.netapp.io but in your list it is just netapp.io/trident, so when my source pvc has accessmode: RWX and destination PVC also i need it as RWX. Since the provisioner names are different in our cluster compared to the list mentioned in  line#40, we are getting warnings in the status field in migplan and the storage class of destination pvcs are getting overridden to default storageclasses with accessmode RWO. I can see you discussed somewhat similar issue to be handled via configmap. has this been implemented? if so, can you share me the details of configmap name and the data it should comprise of. so that we can create the configmap in respectivie namespace inorder to prevent 1. storageclass getting overridden and 2. accessmode is picking the right one instead of default RWO.